dipy/reconst/mdki.py

```diff
@@ -0,0 +1,417 @@
+
+        Returns
+        ---------
+        mk : ndarray
+            Calculated Spherical Mean diffusitivity.
+
+        References
+        ----------
+        .. [1] Henriques, R.N., Correia, M.M., 2017. Interpreting age-related
+               changes based on the mean signal diffusion kurtosis. 25th Annual
+               Meeting of the ISMRM; Honolulu. April 22-28
+        """
+        return self.model_params[..., 1]
+
+    def predict(self, gtab, S0=1.):
+        r"""
+        Given a mean spherical diffusion kurtosis model fit, predict the signal
+        on the vertices of a sphere
+
+        Parameters
+        ----------
+        gtab : a GradientTable class instance
+            This encodes the directions for which a prediction is made
+
+        S0 : float array
+           The mean non-diffusion weighted signal in each voxel. Default:
+           The fitted S0 value in all voxels if it was fitted. Otherwise 1 in
+           all voxels.
+
+        Returns
+        --------
+        S : (..., N) ndarray
+            Simulated mean signal based on the mean spherical kurtosis model
+
+        Notes
+        -----
+        The predicted signal is given by:
+        $MS(b) = S_0 * exp(-bMD + 1/6 b^{2} MD^{2} MK)$, where MD is the
+        mean spherical diffusivity and mean spherical kurtosis.
+
+        References
+        ----------
+        .. [1] Henriques, R.N., Correia, M.M., 2017. Interpreting age-related
+               changes based on the mean signal diffusion kurtosis. 25th Annual
+               Meeting of the ISMRM; Honolulu. April 22-28
+        """
+        return mdki_prediction(self.model_params, gtab, S0=S0)
+
+
+def wls_fit_mdki(design_matrix, msignal, ng, mask=None,
+                 min_signal=MIN_POSITIVE_SIGNAL, return_S0_hat=False):
+    r"""
+    Fits the mean spherical diffusion kurtosis imaging based on a weighted
+    least square solution [1]_.
+
+    Parameters
+    ----------
+    design_matrix : array (nub, 3)
+        Design matrix holding the covariants used to solve for the regression
+        coefficients of the mean spherical diffusion kurtosis model. Note that
+        nub is the number of unique b-values
+    msignal : ndarray ([X, Y, Z, ..., nub])
+        Mean signal along all gradient direction for each unique b-value
+        Note that the last dimension should contain the signal means and nub
+        is the number of unique b-values.
+    ng : ndarray(nub)
+        Number of gradient directions used to compute the mean signal for
+        all unique b-values
+    mask : array
+        A boolean array used to mark the coordinates in the data that
+        should be analyzed that has the shape data.shape[:-1]
+    min_signal : float, optional
+        Voxel with mean signal intensities lower than the min positive signal
+        are not processed. Default: 0.0001
+    return_S0_hat : bool
+        Boolean to return (True) or not (False) the S0 values for the fit.
```

Comment:
I find this sentence (and others like it) unclear. Why not a simpler "If True, also return S0 values for the fit"?